flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java

```diff
@@ -0,0 +1,153 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.formats.json;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.formats.common.TimestampFormat;
+import org.apache.flink.formats.json.JsonOptions;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.util.Collector;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.table.api.DataTypes.DECIMAL;
+import static org.apache.flink.table.api.DataTypes.FIELD;
+import static org.apache.flink.table.api.DataTypes.INT;
+import static org.apache.flink.table.api.DataTypes.ROW;
+import static org.apache.flink.table.api.DataTypes.STRING;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for {@link ChangelogJsonSerializationSchema} and {@link
+ * ChangelogJsonDeserializationSchema}.
+ */
+public class ChangelogJsonSerDeDecimalTest {
+
+    private static final RowType SCHEMA =
+            (RowType)
+                    ROW(
+                                    FIELD("id", INT().notNull()),
+                                    FIELD("name", STRING()),
+                                    FIELD("money", DECIMAL(20, 2)))
+                            .getLogicalType();
+
+    @Test
+    public void testDecimalSerializationDeserialization() throws Exception {
+        List<String> lines = readLines("changelog-json-data2.txt");
+        ChangelogJsonDeserializationSchema deserializationSchema =
+                new ChangelogJsonDeserializationSchema(
+                        SCHEMA, InternalTypeInfo.of(SCHEMA), false, TimestampFormat.SQL);
+
+        deserializationSchema.open(null);
+        SimpleCollector collector = new SimpleCollector();
+        for (String line : lines) {
+            deserializationSchema.deserialize(line.getBytes(StandardCharsets.UTF_8), collector);
+        }
+
+        List<String> expected =
+                Arrays.asList(
+                        "+I(112,liaooo,10.00)",
+                        "-U(112,liaooo,10.00)",
+                        "+U(112,liaooo,1000000.00)",
+                        "-U(112,liaooo,1000000.00)",
+                        "+U(112,liaooo,9999999.50)");
+        List<String> actual =
+                collector.list.stream().map(Object::toString).collect(Collectors.toList());
+        assertEquals(expected, actual);
```

Comment:
We should add a deserialization test data for scientific notation？

Reply:
i refined the relevant test cases and comments, review again.